### PR TITLE
Sparse robust pca

### PR DESCRIPTION
**Pull request by scopatz**
Like the other sparse PRs, this builds on #84.  This enables `robust_pca()` to work with the sparse interface. As mentioned in #88, this may not be the most efficient thing to do because the low rank part is necessarily dense (even though it is stored sparsely). However, the sparse part does end up being sufficiently sparse that it makes sense for that tensor.  

An example of this [is available here](https://github.com/Quansight/tensorly-notebooks/blob/sparse/07_pydata_sparse_backend/sparse-robust-pca-demo.ipynb).

Additionally, this currently needs the master version of PyData/Sparse for the `COO.dot()` method. This should be released soon.

### CODE REVIEW

File: tensorly/decomposition/robust_decomposition.py [hunk 1]
```diff
@@ -90,7 +90,10 @@ def robust_pca(X, mask=None, tol=10e-7, reg_E=1, reg_J=1,
     for iteration in range(n_iter_max):
 
         for i in range(T.ndim(X)):
-            J[i] = fold(svd_thresholding(unfold(D, i) + unfold(L[i], i)/mu, reg_J/mu), i, X.shape)
+            svd_thresh = svd_thresholding(unfold(D, i) + unfold(L[i], i)/mu, reg_J/mu)
+            if not T.is_tensor(svd_thresh):
+                svd_thresh = T.tensor(svd_thresh)
```
Comment (JeanKossaifi) [inline tensorly/decomposition/robust_decomposition.py]: When/why would this happen?
It seems to me that svd_thresholding always returns a tensor.

Comment (JeanKossaifi) [reply tensorly/decomposition/robust_decomposition.py]: @asmeurer I think that's the only change to the actual Robust PCA here, the rest is already added in other PRs

Comment (asmeurer) [reply tensorly/decomposition/robust_decomposition.py]: CC @scopatz

**Comment by asmeurer** [comment]
I don't know for sure but I think if you [resolve the merge conflicts](https://github.com/tensorly/tensorly/pull/90/conflicts) the diff on GitHub will clean up.

**Comment by JeanKossaifi** [comment]
Merging this. Thanks @asmeurer and @scopatz!

**Pull request merged**